components/docusign/sources/envelope-sent-or-complete/envelope-sent-or-complete.js

```diff
@@ -0,0 +1,73 @@
+const docusign = require("../../docusign.app.js");
+
+module.exports = {
+  key: "docusign-envelope-sent-or-complete",
+  name: "Envelope Sent or Complete",
+  description:
+    "Emits an event when an envelope status is set to sent or complete",
+  version: "0.0.2",
+  dedupe: "unique",
+  props: {
+    docusign,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    account: {
+      propDefinition: [
+        docusign,
+        "account",
+      ],
+    },
+  },
+  methods: {
+    _getLastEvent() {
+      return this.db.get("lastEvent");
+    },
+    _setLastEvent(lastEvent) {
+      this.db.set("lastEvent", lastEvent);
+    },
+    monthAgo() {
+      const monthAgo = new Date();
+      monthAgo.setMonth(monthAgo.getMonth() - 1);
+      return monthAgo;
+    },
+    generateMeta({
+      envelopeId: id, emailSubject: summary,
```

Comment:
The `id` needs to be composed from the envelope ID and the envelope status, so that the deduplication algorithm won't block the completed event if/when emitted.

Reply:
Updated to include status in the id